app/src/main/java/com/mybus/activity/MainActivity.java

```diff
@@ -519,8 +526,8 @@ private void cancelProgressDialog() {
      * Clears all markers and polyline for a previous route
      */
     private void clearBusRouteOnMap() {
-        if (mCurrentMapBusRoad != null) {
-            mCurrentMapBusRoad.clearBusRoadFromMap();
+        if (mViewPagerAdapter != null && mViewPager != null && mViewPagerAdapter.getItem(mViewPager.getCurrentItem()) != null) {
```

Comment:
In order to make the code more readable and reusable you should refactor the 'if conditions' that are checking multiple null values on mViewPagerAdapter and mViewPager. You can define and use a method like this one:

`private boolean isViewPagerAdapterPresent() {
  return (mViewPagerAdapter != null && mViewPager != null && mViewPagerAdapter.getItem(mViewPager.getCurrentItem()) != null );	
} `